### ISSUE TITLE
builtins.property not accepted as a valid type for subclassing

### ISSUE THREAD

**Issue by beckjake** [python/mypy#1529]
mypy doesn't like it when I subclass the builtin class `property`:

``` python
class blah(property):
    pass
```

Running mypy 0.4.1 or 9e4cfc8 against that file, I get:
`test.py:1: error: Invalid type "builtins.property"`

But you totally can subclass property and build yourself a customized property decorator. 

Looking at some debugger statements it seems like the type has been classified as not being an instance of TypeInfo. I'm afraid I can't offer a PR to fix this because I have no idea where that should be happening.

**Comment by gvanrossum** [comment]
Yeah, mypy probably only special-cases when the decorator is exactly
'builtins.property'. :-(

Do you have a work-around?

**Comment by beckjake** [comment]
I thought that might be possible too, but changing to "builtins.property" also failed. The only place properties really seem to get handled is in SemanticAnalyzer.visit_decorator, but that never happens since this isn't a decorator.

I did come up with a work-around of just implementing my own descriptor, which works fine and probably is what I should have done in the first place, I just found it an odd curiosity.
